solr/core/src/java/org/apache/solr/cloud/overseer/CollectionMutator.java

```diff
@@ -164,6 +167,23 @@ public ZkWriteCommand modifyCollection(final ClusterState clusterState, ZkNodePr
 
   }
 
+  private DocCollection updateReplicas(DocCollection coll, PerReplicaStates prs) {
+    //we are disabling PRS. Update the replica states
+    Map<String, Slice> modifiedSlices = new HashMap<>();
+    coll.forEachReplica((s, r) -> {
+      PerReplicaStates.State prsState = prs.states.get(r.getName());
+      if(prsState != null && prsState.state != r.getState()) {
+        Slice slice = modifiedSlices.getOrDefault(r.slice, coll.getSlice(r.slice));
+        modifiedSlices.put(r.slice,
```

Comment:
@noblepaul @chatman It seems to me we need to update leader property here. Please let me know what you think about this?
```
noble/prs-scratch:~/work/fullstory/fs-solr/lucene-solr$ git diff
diff --git a/solr/core/src/java/org/apache/solr/cloud/overseer/CollectionMutator.java b/solr/core/src/java/org/apache/solr/cloud/overseer/CollectionMutator.java
index 5a32da6c77..e672be993d 100644
--- a/solr/core/src/java/org/apache/solr/cloud/overseer/CollectionMutator.java
+++ b/solr/core/src/java/org/apache/solr/cloud/overseer/CollectionMutator.java
@@ -174,8 +174,12 @@ public class CollectionMutator {
       PerReplicaStates.State prsState = prs.states.get(r.getName());
       if(prsState != null && prsState.state != r.getState()) {
         Slice slice = modifiedSlices.getOrDefault(r.slice, coll.getSlice(r.slice));
+        Replica replica = new ReplicaMutator(cloudManager).setState(r, prsState.state.toString());
+        if (prsState.isLeader) {
+          replica = new ReplicaMutator(cloudManager).setLeader(replica);
+        }
         modifiedSlices.put(r.slice,
-            slice.copyWith(new ReplicaMutator(cloudManager).setState(r, prsState.state.toString())));
+            slice.copyWith(replica));
       }
     });
     if(!modifiedSlices.isEmpty()) {
```